frontend/src/metabase/dashboard/components/PublicLinkPopover/PublicLinkPopover.unit.spec.tsx

```diff
@@ -0,0 +1,239 @@
+      setup({ isAdmin: false });
+
+      expect(
+        screen.queryByText("Remove this public link"),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  describe("when creating public links", () => {
+    it("should call createPublicLink when uuid is null and isOpen is true", () => {
+      const { createPublicLink } = setup({ hasUUID: false });
+
+      expect(createPublicLink).toHaveBeenCalledTimes(1);
+    });
+
+    it("should not call createPublicLink when isOpen is false", () => {
+      const { createPublicLink } = setup({ isOpen: false, hasUUID: false });
+
+      expect(createPublicLink).not.toHaveBeenCalled();
+    });
+
+    it("should call createPublicLink when uuid is null and the popover is opened", async () => {
+      const { createPublicLink } = setup({ hasUUID: false, isOpen: false });
+
+      userEvent.click(screen.getByTestId("target"));
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+      expect(createPublicLink).toHaveBeenCalledTimes(1);
+    });
+
+    it("should not call createPublicLink when uuid is not null and the popover is opened", async () => {
+      const { createPublicLink } = setup({ hasUUID: true, isOpen: false });
+
+      userEvent.click(screen.getByTestId("target"));
+
+      expect(createPublicLink).not.toHaveBeenCalled();
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+    });
+  });
+
+  describe("when deleting public links", () => {
+    it("should call deletePublicLink and onClose when `Remove this public link` is clicked", () => {
+      const { deletePublicLink, onClose } = setup({
+        hasUUID: true,
+        isOpen: true,
+        isAdmin: true,
+      });
+
+      userEvent.click(screen.getByText("Remove this public link"));
+
+      expect(deletePublicLink).toHaveBeenCalledTimes(1);
+      expect(onClose).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("when copying the link", () => {
+    it("should allow admins to copy the link to the clipboard", async () => {
+      setup({ hasUUID: true, isOpen: true });
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+
+      userEvent.click(screen.getByLabelText("copy icon"));
+
+      expect(await screen.findByText("Copied!")).toBeInTheDocument();
+    });
+
+    it("should allow non-admins to copy the link to the clipboard", async () => {
+      setup({ hasUUID: true, isOpen: true, isAdmin: false });
+
+      expect(await screen.findByText("sample-public-link")).toBeInTheDocument();
+
+      userEvent.click(screen.getByLabelText("copy icon"));
+
+      expect(await screen.findByText("Copied!")).toBeInTheDocument();
+    });
+  });
+
+  describe("when appending extensions to the link", () => {
+    it("should append the extension to the link when the extension is clicked on", async () => {
+      setup({ hasUUID: true, isOpen: true, extensions: ["csv"] });
```

Comment:
I don't think you need to add an option for `extensions`. It would make more sense to add an option whether or not this is a dashboard or a question sharing popover, and if it's a question sharing popover we always pass `["csv", "xlsx", "json"]`. Do you think that'll be easier to understand?

Reply:
I'm not so sure on this one - I think using `extensions` is both more explicit and easier to understand than passing a dashboard or question and then adding extensions based on that. It would be another layer to think about imo

Reply:
It's more explicit yes. But that kind of tailors we allow passing any `extensions` wherever `PublicLinkPopover` is used.

Now the way I see it is that we only have 2 places this `PublicLinkPopover` is being used, and only one of them (questions) can contain `extensions` and extensions aren't dynamic either they're either CSV, XLSX, or JSON.

These tests also play a very important role in the API design, do we really design this component to be able to accept `extensions` dynamically, or do we just want to make it very strict and say if it's rendered for questions, we will pass these 3 extensions, but if it's rendered for dashboard, we don't pass any extensions.

There is no right or wrong answer to this question, but to me, it's a matter of which one is more appropriate at that point in time.

My argument would be it's been like this for I don't know how long and I don't see we will display extensions anywhere else outside questions anytime soon.